### DISCUSSION TITLE
Bad_TypeMismatch problem for help

### DISCUSSION THREAD

**Discussion by linjunli01** [eclipse-milo/milo#1074] [General]
i connect a plc successfully,and it could read the variable,but failed to write it and turn back "Bad_TypeMismatch".
here is my code,is there any help i could get?

        Variant v = new Variant(5);
        DataValue value1 = new DataValue(v, null, null);
        List<StatusCode> statusCodes = opcUaClient.writeValues(nodeId2, ImmutableList.of(value1)).get();
        System.out.println(statusCodes.get(0).isGood());
        System.out.println(statusCodes.get(0));

**Comment by kevinherron** [comment]
The server is probably rejecting the write because you didn't provide a value of the correct DataType.

If you don't know the DataType already you need to read the DataType attribute first to figure it out.
